main/observing/observation_run.py

```diff
@@ -43,17 +57,17 @@ def run_ticket(self, ticket):
 
         for i in range(len(ticket.filter)):
             self.take_images(ticket.name, ticket.num, ticket.exp_time,
-                             list(ticket.filter[i]), ticket.end_time, self.image_directory)
-
+                             list(ticket.filter)[i], ticket.end_time, self.image_directory)
 
-    def take_images(self, name, num, exp_time, filter, end_time, path):
-        num_filters = len(filter)
 
+    def take_images(self, name, num, exp_time, current_filter, end_time, path):
+    
+        image_num = 1
         for i in range(num):
             if end_time <= datetime.datetime.now():
                 print("The observations end time of {} has passed.  "
                       "Stopping observation of {}.".format(end_time, name))
-
-            current_filter = filter[num_filters % i]
```

Comment:
So this idea was correct but my implementation is wrong.  It should be filter[i % num_filters]

Reply:
This touches on how we wanted the observation ticket to be read and executed.  The details of which I don't think have been passed on to you.